lib/getUniqueId.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+module.exports = getUniqueId;
+
+// Given a prefix for a new ID, checks the json for matching prefixes throughout and generates a unique ID.
+function getUniqueId(gltf, prefix) {
+    var prefixMatches = [];
+
+    // Check each top level object in the gltf for instances of the prefix
+    var topLevelObjectIds = Object.keys(gltf);
```

Comment:
Not all top-level objects will have properties that are ids, e.g., `asset`.  Refer to the list I provided in https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/118#issuecomment-229408937 and also used here: https://github.com/AnalyticalGraphicsInc/gltf-pipeline/blob/master/lib/removeUnused.js